### DISCUSSION TITLE
Conditional  `component` gives error

### DISCUSSION THREAD

**Discussion by Toshinaki** [mantinedev/mantine#1292] [Q&A] [answered]
I have a list of items, and some of them might have an url.

To display them in a list, I tried:
```
import { NavLink } from "react-router-dom";

...

<ul>
{items.map(item => 
  <li>
    <Box component={item.url ? NavLink : "p"} ...
```

And vscode complains:
```
Type '"li" | ForwardRefExoticComponent<NavLinkProps & RefAttributes<HTMLAnchorElement>>' is not assignable to type '"li" | undefined'.
  Type 'ForwardRefExoticComponent<NavLinkProps & RefAttributes<HTMLAnchorElement>>' is not assignable to type '"li"'.ts(2322)
[Polymorphic.d.ts(6, 5): ]()The expected type comes from property 'component' which is declared here on type 'IntrinsicAttributes & _BoxProps & ComponentProp<"li"> & Omit<Pick<DetailedHTMLProps<LiHTMLAttributes<HTMLLIElement>, HTMLLIElement>, "key" | keyof LiHTMLAttributes<...>>, "component" | keyof _BoxProps> & { ...; }'
```
![image](https://user-images.githubusercontent.com/10347466/164441153-a8200459-1732-44ce-a832-26bf1de64ba4.png)

Is it impossible to do this with Mantine?

BTW, the react version is 18, and Mantine the latest version.

**Comment by rtivital** [comment, marked as answer]
Use `as any` or `as const` if you have type errors

**Reply by Toshinaki** [reply]
Yes `component={(item.url ? NavLink : "p") as any}` did the trick, but **`any`** does not feel so right... since I'm using typescript...

Any chance we get some update for this?

**Reply by rtivital** [reply]
It is not possible to achieve that with typescript
